src/html/extractors/factories/embeddedAttributeJs.ts

```diff
@@ -1,24 +1,31 @@
+import { Attribute } from 'parse5';
 import { JsParser } from '../../../js/parser';
 import { Validate } from '../../../utils/validate';
 import { Element, IHtmlExtractorFunction, Node } from '../../parser';
-/**
- * embeddedAttrJsExtractor extractor messages from element attribute,
- * but treat them as embedded js:
- *
- *     <span :title="_xn('ctx', 'msgid', 'plural', n, args)">See my title</span>
- */
 
-export function embeddedAttrJsExtractor(selector: string, jsParser: JsParser): IHtmlExtractorFunction {
-    Validate.required.nonEmptyString({ selector });
+export type AttributePredicate = (attribute: Attribute) => boolean;
+
+export function embeddedAttributeJsExtractor(jsParser: JsParser, filter: null | string | AttributePredicate): IHtmlExtractorFunction {
     Validate.required.argument({ jsParser });
+    let test: AttributePredicate = attr => true;
+    if (typeof filter === 'string') {
+        const namePattern = filter;
+        test = (attr: Attribute) => {
+            if (attr.name.match(namePattern)) { return true; }
```

Comment:
If you want to use regex, you should adjust the parameter type. Converting a string to a regex can be misleading to the consumer since they might not know that they have to watch out for characters with special meaning in regex.

Also in this case it makes more sense to use `RegExp.prototype.test` since you're not interested in the match.